src/tools/kdb/complete.cpp

```diff
@@ -102,6 +106,17 @@ void CompleteCommand::addMountpoints (KeySet & ks, const Key root)
 	printWarnings (cerr, mountpointPath);
 }
 
+void CompleteCommand::addNamespaces (map<Key, pair<int, int>> & hierarchy)
+{
+	// Currently assumed to be fixed, and they are on level 0
+	const string namespaces[] = { "spec/", "proc/", "dir/", "user/", "system/" };
```

Comment:
Would be great if we do not need to hardcode it or at least get a warning/error if we add an namespace. See `keyGetNamespace`.

To get all namespaces might be a good addition in the c++ binding (or at least in libtools).